pkg/proc/gdbserial/gdbserver.go

```diff
@@ -1180,14 +1141,7 @@ func (p *gdbProcess) restartWorker(cctx *proc.ContinueOnceContext, pos string) (
 
 	var err error
 	if p.conn.isUndoServer {
-		switch pos {
-		case "start":
-			err = p.conn.restart("")
-		case "end":
-			_, err = p.conn.undoCmd("goto_record_mode")
-		default:
-			err = p.conn.restart(pos)
-		}
+		err = p.undoSession.travelToTime(p, pos)
```

Comment:
Similarly here. The invariant seems to be that `p.conn.isUndoServer` if and only if `p.undoSession != nil`. Is that right? If so, could we drop the former and use the latter instead? (Not sure if this is possible.)